integration_tests/e2e/prisonerList/prisonerList.cy.ts

```diff
@@ -36,48 +35,121 @@ context(`Display the prisoner list screen`, () => {
   it('should be able to navigate directly to the prisoner list page', () => {
     // Given
     cy.signIn()
+    const expectedResultCount = prisonerSearchSummaries.length
 
     // When
     cy.visit('/')
 
     // Then
     const prisonerListPage = Page.verifyOnPage(PrisonerListPage)
-    prisonerListPage.hasResultsDisplayed()
+    prisonerListPage.hasResultsDisplayed(expectedResultCount)
   })
 
-  it('should search for non existent prisoner and get zero results', () => {
-    // Given
-    cy.signIn()
-    cy.visit('/')
-    const prisonerListPage = Page.verifyOnPage(PrisonerListPage)
+  describe('filtering', () => {
+    it('should filter for non existent prisoner and get zero results', () => {
+      // Given
+      cy.signIn()
+      cy.visit('/')
+      const prisonerListPage = Page.verifyOnPage(PrisonerListPage)
 
-    // When
-    prisonerListPage //
-      .setNameFilter('some non existent search term')
-      .applyFilters()
+      // When
+      prisonerListPage //
+        .setNameFilter('some non existent search term')
+        .applyFilters()
 
-    // Then
-    prisonerListPage.hasNoResultsDisplayed()
+      // Then
+      prisonerListPage.hasNoResultsDisplayed()
+    })
+
+    it('should filter for prisoners named John', () => {
+      // Given
+      cy.signIn()
+      cy.visit('/')
+      const prisonerListPage = Page.verifyOnPage(PrisonerListPage)
+
+      const numberOfPrisonersNamedJohn = prisonerSearchSummaries.filter(
```

Comment:
There will be some John's in the `prisonerSearchSummaries`, but given that it was randomly generated (725 records) we don't know how many of them have John in their name, so we need to work it out